packages/data/src/reviews/reducer.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Internal dependencies
+ */
+import TYPES from './action-types';
+
+const reducer = (
+	state = {
+		reviews: {},
+		errors: {},
+		data: {},
+	},
+	{ type, query, reviews, totalCount, error }
+) => {
+	switch ( type ) {
+		case TYPES.UPDATE_REVIEWS:
+			const ids = [];
+			const nextReviews = reviews.reduce( ( result, review ) => {
+				ids.push( review.id );
+				result[ review.id ] = review;
+				return result;
+			}, {} );
+			state = {
+				...state,
+				reviews: {
+					...state.reviews,
+					[ JSON.stringify( query ) ]: { data: ids, totalCount },
+				},
+				data: {
+					...state.data,
+					...nextReviews,
+				},
+			};
+			break;
```

Comment:
Again personal preference (although I think in this case it would be more succinct):

We probably don't need to mutate the state variable at all in each `case` and then we can remove the `break;` statements and the end return statement. Instead I think it's a bit more idiomatic/succinct to return the state from the case. I also personally like having `{}` around the case but I'm not sure if our lint supports this (it does also let you declare the same var name twice between case statements which can be useful). Anyway, so without the `{}` would look like:

```
case TYPES.UPDATE_REVIEWS:
			const ids = [];
			const nextReviews = reviews.reduce( ( result, review ) => {
				ids.push( review.id );
				result[ review.id ] = review;
				return result;
			}, {} );
			return {
				...state,
				reviews: {
					...state.reviews,
					[ JSON.stringify( query ) ]: { data: ids, totalCount },
				},
				data: {
					...state.data,
					...nextReviews,
				},
			};
```

we lose the return statement at the end and we lose all the breaks. It can also be nice to provide a `default` case that just returns state untouched.

Reply:
Cool, done